src/test/resources/com/jpinpoint/perf/lang/kotlin/ruleset/common/xml/AvoidMultipleConcatStatements.xml

```diff
@@ -0,0 +1,206 @@
+    }
+
+    fun testMultipleConcatDefect4() {
+        var log = ""
+        var i = 0
+        val values = listOf(1, 2, 3)
+        while (i++ < 3) {
+            log = log + values[0]
+            log = log + values[1] // bad 51
+        }
+    }
+
+    fun testMultipleConcatCorrect4() {
+        var log = 0
+        var i = 0
+        while (i++ < 3) {
+            val values = listOf(1, 2, 3)
+            log = log + values[0]
+            log = log + values[1]
+        }
+    }
+
+    fun testMultipleConcatDefect5() {
+        var logStatement = ""
+        val values = listOf("tic", "tac", "toe")
+        logStatement = logStatement + values[0]
+        logStatement += values[1] // bad 69
+    }
+
+    fun testMultipleConcatCorrect5() {
+        var logStatement = ""
+        val values = listOf("tic", "tac", "toe")
+        logStatement = logStatement + values[0] + values[1]
+    }
+
+    fun testMultipleConcatDefect6() {
+        var log = someString();
+        var i = 0
+        while (i++ < 3) {
+            val values = listOf(1, 2, 3)
+            log = log + values[0]
+            log = log + values[1] //bad 84
+        }
+    }
+
+    fun testMultipleConcatCorrect6() {
+        var log = someInt("");
+        var i = 0
+        while (i++ < 3) {
+            val values = listOf(1, 2, 3)
+            log = log + values[0]
+            log = log + values[1]
+        }
+    }
+
+    fun testMultipleConcatDefect7() {
+        var logStatement: String = someUnknown();
+        val values = listOf("tic", "tac", "toe")
+        logStatement += values[0]
+        logStatement += values[1] // bad 102
+    }
+
+    fun testMultipleConcatCorrect7() {
+        var logStatement: String = someUnknown();
+        val values = listOf("tic", "tac", "toe")
+        logStatement += values[0] + values[1]
+    }
+
+    fun testMultipleConcatDefect8() {
+        var logStatement = "{"
+        val values = listOf("tic", "tac", "toe")
+        logStatement += "$values[0], "
+        logStatement += "$values[1]}" // bad 115
+    }
+
+    fun testMultipleConcatCorrect8() {
+        var logStatement = "{"
+        val values = listOf("tic", "tac", "toe")
+        logStatement += "$values[0], $values[1]}"
+    }
+
+    fun testMultipleConcatDefect9(first: String) {
+        var logStatement = first;
+        val values = listOf("tic", "tac", "toe")
+        logStatement += values[0]
+        logStatement += values[1] // bad 128
+    }
+
+    fun testMultipleConcatCorrect9(first: Int) {
+        var logStatement = first;
+        val values = listOf(1, 2, 3)
+        logStatement += values[0]
+        logStatement += values[1]
+    }
+
+    fun testMultipleConcatDefect10() {
+        var logStatement = someString();
+        val values = listOf("tic", "tac", "toe")
+        logStatement += values[0]
+        logStatement += values[1] // bad 142
+    }
+
+    fun testMultipleConcatCorrect10() {
+        var logStatement = someInt("");
```

Comment:
why a test case like this? what is intention of `someInt()` vs `someString()`?

Reply:
Ah, probably to lookup the return value of the method (in same class only)

Reply:
exactly